Plugins/BTCPayServer.Plugins.ShopifyPlugin/Controllers/UIShopifyV2Controller.cs

```diff
@@ -28,14 +27,16 @@
 using BTCPayServer.Plugins.ShopifyPlugin.Clients;
 using BTCPayServer.Plugins.ShopifyPlugin.ViewModels;
 using Microsoft.Extensions.Configuration;
+using Microsoft.AspNetCore.Cors;
 
 namespace BTCPayServer.Plugins.ShopifyPlugin;
 
 [Authorize(AuthenticationSchemes = AuthenticationSchemes.Cookie, Policy = Policies.CanViewProfile)]
 [AutoValidateAntiforgeryToken]
 public class UIShopifyV2Controller : Controller
 {
-	private readonly StoreRepository _storeRepo;
+    private readonly string[] _keywords = new[] { "bitcoin", "btc", "btcpayserver", "btcpay server" };
```

Comment:
Do not duplicate code.

Instead, just create a method like `static bool IsBTCPayServerGateway(string gateway)` in `ShopifyService`.

And refactor the `var baseParentTransaction = existingShopifyOrderTransactions` to use this instead of the current stuff, then use `ShopifyService.IsBTCPayServerGateway` in your controller.